linalg_ops/convolution/generate_e2e_conv2d_tests.py

```diff
@@ -0,0 +1,739 @@
+
+
+# Enumerates ways to initialize kernel buffer contents.
+@enum.unique
+class KernelGenerator(enum.Enum):
+    ZERO = "zero"  # Fill with zeros
+    RANDOM = "random"  # Fill with (deterministic) pseudorandom values.
+
+
+# TODO: Add more input layouts as needed. The layout determines the dim of input and kernel.
+@enum.unique
+class InputLayout(enum.Enum):
+    NCHW = "nchw"
+    NHWC = "nhwc"
+
+
+# TODO: Add more kernel layouts as needed.
+@enum.unique
+class KernelLayout(enum.Enum):
+    FCHW = "fchw"
+    HWCF = "hwcf"
+
+
+# Describes the shape of a tensor conv2d in the usual convention:
+# the input is {n}x{c}x{h}x{w}, the kernel is {f}x{c}x{kh}x{kw}, the accumulator/result is
+# {n}x{f}x{oh}x{ow}.
+# The extra `accumulate` boolean tells whether the conv2d is accumulating into
+# an existing accumulator (C += A * B) or just overwriting the result
+# (C = A * B).
+@dataclasses.dataclass
+class TestShape:
+    n: int
+    c: int
+    h: int
+    w: int
+    kh: int
+    kw: int
+    f: int
+    accumulate: bool
+
+
+# Attributes for the linalg.conv2d operation.
+@dataclasses.dataclass
+class ConvAttrs:
+    STRIDE: typing.Tuple[int, int] = (1, 1)
+    DILATION: typing.Tuple[int, int] = (1, 1)
+
+
+# Returns the list of TestShape's to use for the collection of shapes
+# identified by shapes_id.
+def get_test_shapes(shapes_id: ShapesId):
+    # Notes:
+    # 1. Be conservative in adding more shapes, as that can increase both the
+    #    build and execution latency of tests. The build latency is nearly the
+    #    same for all shapes, while execution latency grows linearly with
+    #    n*f*ow*oh*kh*kw.
+
+    if shapes_id == ShapesId.SMALL:
+        return [
+            TestShape(n=1, c=1, h=1, w=1, kh=1, kw=1, f=1, accumulate=True),
+            TestShape(n=1, c=1, h=16, w=16, kh=2, kw=2, f=1, accumulate=True),
+            TestShape(n=2, c=2, h=32, w=32, kh=3, kw=3, f=2, accumulate=True),
+        ]
+    if shapes_id == ShapesId.MEDIUM:
+        return [
+            TestShape(n=2, h=32, w=32, c=32, kh=3, kw=3, f=64, accumulate=True),
+        ]
+    if shapes_id == ShapesId.LARGE:
+        return [
+            TestShape(n=2, c=4, h=128, w=128, kh=3, kw=3, f=8, accumulate=True),
+            TestShape(n=2, c=3, h=128, w=128, kh=3, kw=3, f=12, accumulate=True),
+        ]
+    if shapes_id == ShapesId.GPU_LARGE:
+        return [TestShape(n=1, c=64, h=130, w=130, kh=3, kw=3, f=64, accumulate=True)]
+    
+    raise ValueError(shapes_id)
+
+
+# Returns the list of Dynamicity's to use for the collection of shapes
+# identified by shapes_id.
+def get_dynamicities(shapes_id: ShapesId):
+    if shapes_id == ShapesId.LARGE or shapes_id == ShapesId.GPU_LARGE:
+        return [
+            Dynamicity.STATIC,
+        ]
+    # TODO: Enable dynamic dimensions once the tests start passing.
+    else:
+        return [
+            Dynamicity.STATIC,
+        ]
+    raise ValueError(shapes_id)
```

Comment:
I'm not a fan of tightly coupling the choice of static vs dynamic vs mixed here with the shapes to be tested. With the removal of "GPU shapes" in https://github.com/iree-org/iree-test-suites/pull/22, I just replaced this helper function with an inlined
```python
        for dynamicity in [Dynamicity.DYNAMIC, Dynamicity.STATIC]:
```